receiver/src/main/java/battery/BatteryParamReceiver.java

```diff
@@ -0,0 +1,99 @@
+package battery;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import compute.AverageComputerFunction;
+import compute.MaximumComputerFunction;
+import compute.MinimumComputerFunction;
+
+public class BatteryParamReceiver {
+
+	private BatteryParameter batteryParam = new BatteryParameter();
+	
+	private MinimumComputerFunction minFunction=new MinimumComputerFunction();
+	
+	private MaximumComputerFunction maxFunction = new MaximumComputerFunction();
+	
+	private AverageComputerFunction avgFunction = new AverageComputerFunction();
+	
+	public void readAndCompute(Scanner scanner) throws Exception {
+
+		try {
+			String inputParam = null;
+			int counter = 0;
+			while (scanner.hasNext()) {
+				inputParam = scanner.nextLine();
+				
+				validateInput(inputParam);
+				
+				addToModel(inputParam.split("\\s"));
+				
+				counter++;
+				
+				compute(counter);
+
+			}
+		}
+		catch (Exception e) {
```

Comment:
Avoid catching Exception, because this will catch exceptions that you didn't intend to catch as well. Sometimes that makes debugging more time-consuming.